packages/vscode-extension/src/core/sdksetup/SdkServices.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as https from 'https';
+import * as http from 'http';
+import { homedir } from 'os';
+import { parse } from 'url';
+import { VSTranslationService } from '../../service/VSTranslationService';
+import { EXTENSION_INSTALLATION } from '../../service/TranslationKeys';
+import MessageService from '../../service/MessageService';
+import { validateSdk } from './SdkValidator';
+import { FileSystemService } from '../../util/ExtensionUtil';
+import { SDK_DOWNLOAD_URL, SDK_FILENAME, SUITECLOUD_FOLDER, VSCODE_SDK_FOLDER, sdkPath } from './SdkProperties';
+
+const VALID_JAR_CONTENT_TYPES = ['application/java-archive', 'application/x-java-archive', 'application/x-jar'];
+
+const messageService = new MessageService();
+const translationService = new VSTranslationService();
+const fileSystemService = new FileSystemService();
+
+export async function installIfNeeded() {
+	if (!fs.existsSync(path.join(sdkPath)) || !(await validateSdk(path.join(sdkPath)))) {
+		await install();
+	}
+}
+
+async function install() {
+	const sdkParentDirectory = fileSystemService.createFolder(homedir(), SUITECLOUD_FOLDER);
+	const sdkDirectory = fileSystemService.createFolder(sdkParentDirectory, VSCODE_SDK_FOLDER);
+	const fullURL = `${SDK_DOWNLOAD_URL}/${SDK_FILENAME}`;
+
+	try {
+		var downloadFilePromise = downloadFile(fullURL, sdkDirectory);
+		messageService.showInformationMessage(
+			translationService.getMessage(EXTENSION_INSTALLATION.IN_PROGRESS),
+			downloadFilePromise,
+			`$(sync~spin) ${translationService.getMessage(EXTENSION_INSTALLATION.IN_PROGRESS)}`

```

Comment:
Is this $(sync~spin) a kind of spinner? Can we add it instead of only in this message to all the statusBar messages adding this same string literal to the showStatusBarMessage function of the MessageService.ts?
It will be very cool to have it for all the statusBar messages.